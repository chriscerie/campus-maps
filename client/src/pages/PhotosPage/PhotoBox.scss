.photo-container {
  display: block;
  float: left;
  width: 150px;
  margin: 0 12px 12px 0;
}

.photo-box {
  height: 150px;
  position: relative;
}

.photo-box:hover {
  filter: opacity(80%);
}


.photo-box-img {
  max-width: 100%;
  max-height: 100%;
}

.photo-box .photo-box-img {
  border-radius: 4px;
  vertical-align: middle;
}

.photo-link {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: white;
  opacity: 0;
}