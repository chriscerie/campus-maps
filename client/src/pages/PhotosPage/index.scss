.photos-page-container {
  min-height: 100vh;
  width: 100%;
  max-width: 960px;
  box-sizing: border-box;
  margin: 0 auto;
  padding: 1em;
  padding-bottom: 36px;
  overflow-y: auto;
  overflow-x: hidden;
}

.header-container {
  margin-top: 18px;
  margin-bottom: 11px;
  border-bottom: 1px solid #e6e6e6;
}

.subheader-container {
  display: table;
  min-width: 100%;
  table-layout: auto;
  margin-bottom: 12px;
}

.subheader-title {
  width: 100%;
  box-sizing: border-box;
  display: table-cell;
  vertical-align: top;
}

.add-photo-button {
  box-sizing: border-box;
  display: table-cell;
  vertical-align: bottom;
  white-space: nowrap;
}

.photos-grid-container {
  margin-right: -12px;
  transform-style: preserve-3d;
  margin-top: 18px;
  *zoom: 1;
  overflow: auto;
}

.photos-grid {
  overflow: auto;
  list-style: none;
  padding: 0;
  margin: 0;
  border: 0;
  max-height: 50em;
  cursor: pointer;
}

.pagination-container {
  border-bottom: 1px solid #e6e6e6;
}
