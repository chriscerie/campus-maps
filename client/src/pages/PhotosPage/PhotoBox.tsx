//import type { UserType } from '../../types/UserType';
import './PhotoBox.scss';

export type PhotoType = {
  _id: string;
  user_id: string;
  photo_src: string;
  // user?: UserType; time posted, etc..
};

const PhotoBox = ({ photo }: { photo: PhotoType }) => {
  console.log(photo);
  console.log(photo.user_id);
  return (
    <li className="photo-container">
      <div className="photo-box">
        <img
          className="photo-box-img"
          src={photo.photo_src}
          alt="location"
          width="226"
          height="226"
        />
      </div>
    </li>
  );
};

export default PhotoBox;
