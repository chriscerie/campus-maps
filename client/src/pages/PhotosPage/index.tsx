/* eslint-disable eqeqeq */
import { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { useParams, Link } from 'react-router-dom';
import { RootState } from '../../reducers';
import { Button } from '@mui/material';
import PhotoBox from './PhotoBox';
import { PhotoType } from './PhotoBox';
import { getLocationInfo } from '../../api/LocationAPI';
import type { LocationType } from '../../types/LocationType';
import CameraAltOutlinedIcon from '@mui/icons-material/CameraAltOutlined';
import './index.scss';

function PhotosPage() {
  const { id } = useParams<{ id: string }>();
  //const [backendPhotos, setBackendPhotos] = useState<Array<PhotoType>>([]);
  const [locationInfo, setLocationInfo] = useState<null | LocationType>(null);
  const mapInstance = useSelector((state: RootState) => state.mapInstance.map);

  useEffect(() => {
    if (mapInstance) {
      getLocationInfo(id, mapInstance).then((res) => {
        if (res) {
          setLocationInfo(res);
        }
      });
    }
  }, [id, mapInstance]);

  //temporary photos for testing
  const backendPhotos: PhotoType[] = new Array<PhotoType>();
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });
  backendPhotos.push({
    _id: '1',
    user_id: 'J',
    photo_src:
      'https://s3-media0.fl.yelpcdn.com/bphoto/jrsxTUcsuV6PDSQp50T-Cw/258s.jpg',
    photo_link:
      '/biz_photos/popping-yolk-cafe-hacienda-heights-2?select=jrsxTUcsuV6PDSQp50T-Cw',
  });

  return (
    <div className="photos-page-container">
      <div className="header-container">
        <h1 className="header-title">
          Photos for {locationInfo && locationInfo.name}
        </h1>
        <div className="subheader-container">
          <h3 className="subheader-title">
            {locationInfo && locationInfo.description}
          </h3>
          <div className="add-photo-button">
            <Button
              variant="outlined"
              color="inherit"
              startIcon={<CameraAltOutlinedIcon />}
              sx={{ textTransform: 'none', fontSize: '1em' }}
            >
              Add Photo
            </Button>
          </div>
        </div>
      </div>
      <div className="photos-grid-container">
        <ul className="photos-grid">
          {backendPhotos.map((currentPhoto: PhotoType) => (
            <PhotoBox key={currentPhoto._id} photo={currentPhoto} />
          ))}
        </ul>
      </div>
    </div>
  );
}

export default PhotosPage;
