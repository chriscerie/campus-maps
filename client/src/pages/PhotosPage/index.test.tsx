import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../reducers';
import PhotosPage from './index';

describe('PhotosPage', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <PhotosPage />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
