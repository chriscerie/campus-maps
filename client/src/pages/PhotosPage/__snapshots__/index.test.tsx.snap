// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotosPage should render successfully 1`] = `
<div
  className="photos-page-container"
>
  <div
    className="header-container"
  >
    <div
      className="subheader-container"
    >
      <a
        href="/loc/undefined"
        onClick={[Function]}
        style={
          Object {
            "color": "#000",
            "textDecoration": "none",
          }
        }
      >
        <h1
          className="subheader-title"
        />
      </a>
      <div
        className="add-photo-button"
      >
        <a
          href="/write-review/undefined"
          onClick={[Function]}
          style={
            Object {
              "color": "#000",
              "textDecoration": "none",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-cduwo-MuiButton-root{font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:5px 15px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border:1px solid rgba(0, 0, 0, 0.23);color:inherit;border-color:currentColor;text-transform:none;font-size:1em;}.css-cduwo-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-cduwo-MuiButton-root:hover{background-color:transparent;}}.css-cduwo-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);border:1px solid rgba(0, 0, 0, 0.12);}",
              }
            }
            data-emotion="css cduwo-MuiButton-root"
          />
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-rgay74-MuiButtonBase-root-MuiButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:5px 15px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border:1px solid rgba(0, 0, 0, 0.23);color:inherit;border-color:currentColor;text-transform:none;font-size:1em;}.css-rgay74-MuiButtonBase-root-MuiButton-root::-moz-focus-inner{border-style:none;}.css-rgay74-MuiButtonBase-root-MuiButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-rgay74-MuiButtonBase-root-MuiButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-rgay74-MuiButtonBase-root-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-rgay74-MuiButtonBase-root-MuiButton-root:hover{background-color:transparent;}}.css-rgay74-MuiButtonBase-root-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);border:1px solid rgba(0, 0, 0, 0.12);}",
              }
            }
            data-emotion="css rgay74-MuiButtonBase-root-MuiButton-root"
          />
          <button
            className="MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonBase-root css-rgay74-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-1d6wzja-MuiButton-startIcon{display:inherit;margin-right:8px;margin-left:-4px;}.css-1d6wzja-MuiButton-startIcon>*:nth-of-type(1){font-size:20px;}",
                }
              }
              data-emotion="css 1d6wzja-MuiButton-startIcon"
            />
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <style
                dangerouslySetInnerHTML={
                  Object {
                    "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
                  }
                }
                data-emotion="css i4bv87-MuiSvgIcon-root"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CameraAltOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4h-3.17L15 2H9L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V6h4.05l1.83-2h4.24l1.83 2H20v12zM12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0 8c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                />
              </svg>
            </span>
            Add Photo
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    className="photos-grid-container"
  >
    <ul
      className="photos-grid"
    />
  </div>
</div>
`;
